ocsci/pytest_customization/ocscilib.py

```diff
@@ -103,3 +103,13 @@ def process_cluster_cli_params(config):
         cluster_path = os.path.join(cluster_dir_parent, cluster_name)
     ocsci_config.ENV_DATA['cluster_name'] = cluster_name
     ocsci_config.ENV_DATA['cluster_path'] = cluster_path
+
+
+def pytest_collection_modifyitems(session, config, items):
+    """
+    Add Polarion ID property to test cases that are marked with one.
+    """
+    for item in items:
+        for marker in item.iter_markers(name="polarion_id"):
+            polarion_id = marker.args[0]
+            item.user_properties.append(("polarion-testcase-id", polarion_id))
```

Comment:
We will need to add one more global required parameter under testsuites to junit file. See the documentation:
https://mojo.redhat.com/docs/DOC-1073077#jive_content_id_polarionprojectid_required

Reply:
Yes, but I would rather not add it for every test case individually.

Reply:
Yeah like @mbukatov said in his top level comment we might need to wrap the output from our tests in a `<testsuites>` element as it doesn't exist in the report by default. The property you linked to needs to be a property of `<testsuites>` not the `<testcase>` so I wouldn't add it in this loop.

Reply:
Ok, I've just dig in to how it's done in rhev and you can see one of generated xunit file which works for polarion importer in RHV automation here: https://rhv-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/4.3/job/rhv-4.3-ge-runner-tier1/81/artifact/xunit_output.xml 

I see that it can be in testusite under properties as you can see here
```xml
<testsuite errors="0" failures="3" name="pytest" skips="42" tests="706" time="68101.387">
    <properties>
        <property name="polarion-custom-isautomated" value="True"/>
        <property name="polarion-testrun-status-id" value="inprogress"/><property name="polarion-custom-plannedin" value="RHV_4_3"/>
        <property name="polarion-user-id" value="ci-user"/>
        <property name="polarion-project-id" value="RHEVM3"/><property name="polarion-response-myproduct" value="rhvm"/>
        <property name="polarion-testrun-id" value="RHV_4_3_tier1_x86_64"/><property name="polarion-custom-arch" value="x8664"/>
    </properties>
    <testcase classname="rhevmtests.compute.sla.cpu_qos.cpu_qos_test.TestQoSAndCpuProfileCRUD" file="rhevmtests/compute/sla/cpu_qos/cpu_qos_test.py" line="60" name="test_a_add_qos" time="11.999502182"><properties><property name="polarion-testcase-id" value="RHEVM-14931"/></properties>
        <system-out>
...output omitted...
```

Reply:
I added a session fixture to handle adding these properties. For now it only adds the `polarion-project-id` but we can update it to add any others we might need.

Reply:
Good.